### ISSUE TITLE
Some 3rd party softwares cause Magento to return an internal error on search page

### ISSUE THREAD

**Issue by olzen90** [clerkio/clerk-magento#34]
When using 3rd party softwares like ManaDev, which inserts filters on the search page, a lookup to Magentos internal SQL database is still made even after activating the module.

This causes Magento to return an error page when a misspell is made, and the word cant be found in the database.

A possible fix is this: 

Find the file
_app/code/core/Mage/CatalogSearch/Model/Fulltext.php_ 

Insert the IF-Statement `if ($query->getId() != '') {}` around the following line `$this->getResource()->prepareResult($this, $queryText, $query);`

It should look like this:
```
if ($query->getId() != '') {
    $this->getResource()->prepareResult($this, $queryText, $query);
}

```

This will cause Magento to ignore if the SQL lookup returns false, so it doesnt give back an error.

**Comment by RBech** [comment]
@olzen90 I'm unable to reproduce this with the manadev extension, would you happen to know of any other extensions known to cause this problem?

**Comment by JacobSMoller** [comment]
Amastys search extension causes the problem

**Comment by JacobSMoller** [comment]
https://amasty.com/improved-layered-navigation.html

**Comment by JacobSMoller** [comment]
Also if you are testing on magento.clerk.io i think the fix we had might be implemented already

**Comment by RBech** [comment]
@JacobSMoller, @olzen90 would you happen to have a screenshot or logs displaying the error? I can't replicate the error with manadev or the amasty extension :/

**Comment by olzen90** [comment]
Hi Rasmus.

Yes I actually found an old screenshot from when this happened to a
customer!

I have attached it here.

The error occurred when you searched for a misspelled word, or any word
that didnt exist in the SQL database.



*Stefan Lorenz Olsen*
Customer Happiness and Success



Kigkurren 8G, 2nd floor
DK-2300 Copenhagen S
Office: +45 70 555 100
Email: slo@clerk.io

<http://t.sigopn03.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs4WJbxzMdVgrddV4CgVQBcsP56dHsxf1bcQyg02?t=https%3A%2F%2Fwww.linkedin.com%2Fcompany%2F5307591&si=5146701110181888&pi=0a80072d-9549-491f-8ed2-c9c7463c948a>
  <https://www.facebook.com/www.Clerk.io>

Learn how Lomax <https://youtu.be/sVPP1EO7WuM> and PostMe
<https://youtu.be/87zV9at-VPc> find success with Clerk.io
and see what we can do for your business <https://clerk.io/review>.

**Comment by RBech** [comment]
Having examined the error this seems to be happening in magento < 1.9 and I believe this was solved by cc25c957b98cb42f5e0895ec1961d27594a2889b I've added a 2.0.7 release implementing this commit and will be closing this for now, please reopen if the commit doesn't resolve the issue :)
